### PR TITLE
ITP-1758 

### PR DESCRIPTION
**Pull request by exagenio**
@shamal-iroshan 
@udara-sinister

### CODE REVIEW

File: src/containers/Signin/Signin.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    const randomNumber = Math.floor(Math.random() * names.length);
+    return names[randomNumber];
+  };
+
+  function clickHandler() {
+    setNickname(randomNameGenerate);
+    setDisabled(false);
+  }
+  function changeHandler(event: { target: { value: any } }) {
+    console.log("changed");
```
Comment (udaraJ08) [inline src/containers/Signin/Signin.tsx]: Remove unnecessary console.logs

File: src/containers/Signin/Signin.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    setNickname(event.target.value);
+    if (event.target.value !== "") {
+      setDisabled(false);
+    } else {
+      setDisabled(true);
+    }
+  }
+
+  function continueHandler(){
+    window.location.href = "/home";
```
Comment (udaraJ08) [inline src/containers/Signin/Signin.tsx]: Use React router do not use window location based routings

**Review by udaraJ08 — APPROVED** [review]

File: src/containers/Signin/Signin.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import EastIcon from "@mui/icons-material/East";
+function Signin() {
+    // set variables for nickname and disabled state
+  const [nickname, setNickname] = useState("");
+  const [isDisabled, setDisabled] = useState(true);
+  
+  const navigate = useNavigate();
+
+  // Generates a random name from a list
+  const randomNameGenerate = () => {
```
Comment (shamal-iroshan) [inline src/containers/Signin/Signin.tsx]: let's maintain a different file to handle these types of utility functions. create a folder name utils and move this function to that file

File: src/containers/Signin/Signin.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      "William",
+      "Sophia",
+    ];
+    const randomNumber = Math.floor(Math.random() * names.length);
+    return names[randomNumber];
+  };
+
+  // Click event handler for the "Random" button
+  function clickHandler() {
+    setNickname(randomNameGenerate);// Set a random name
```
Comment (shamal-iroshan) [inline src/containers/Signin/Signin.tsx:15]: is this working?

**Review by shamal-iroshan — CHANGES_REQUESTED** [review]
fix the commented things

File: src/containers/Signin/Signin.tsx (hunk 4 above)
Comment (exagenio) [reply src/containers/Signin/Signin.tsx:15]: Yes it's working

**Comment by exagenio** [comment]
@shamal-iroshan made the changes

**Review by shamal-iroshan — APPROVED** [review]

**Pull request merged**